libs/sdk-core/src/logger.rs

```diff
@@ -0,0 +1,321 @@
+        .format(|buf, record| {
+            writeln!(
+                buf,
+                "[{} {} {}:{}] {}",
+                Local::now().format("%Y-%m-%d %H:%M:%S%.3f"),
+                record.level(),
+                record.module_path().unwrap_or("unknown"),
+                record.line().unwrap_or(0),
+                record.args()
+            )
+        });
+    if let Some(target) = target {
+        builder.target(target);
+    }
+    if let Some(filter_level) = filter_level {
+        builder.filter_level(filter_level);
+    }
+    builder.build()
+}
+
+/* Dart */
+
+static INIT_DART_LOGGER: Once = Once::new();
+
+pub fn init_dart_logger(filter_level: Option<BindingLevelFilter>) {
+    INIT_DART_LOGGER.call_once(|| {
+        let filter_level = get_filter_level(filter_level);
+
+        assert!(
+            filter_level <= STATIC_MAX_LEVEL,
+            "Should respect STATIC_MAX_LEVEL={:?}, which is done in compile time. level{:?}",
+            STATIC_MAX_LEVEL,
+            filter_level
+        );
+
+        let env_logger = init_env_logger(Some(Target::Stdout), Some(filter_level));
+
+        let dart_logger = DartLogger { env_logger };
+        set_boxed_logger(Box::new(dart_logger))
+            .unwrap_or_else(|_| error!("Log stream already created."));
+        set_max_level(filter_level);
+    });
+}
+
+lazy_static! {
+    static ref DART_LOGGER_STREAM_SINK: RwLock<Option<StreamSink<LogEntry>>> = RwLock::new(None);
+}
+
+pub struct DartLogger {
+    env_logger: Logger,
+}
+
+impl DartLogger {
+    pub fn set_stream_sink(stream_sink: StreamSink<LogEntry>) {
+        let mut guard = DART_LOGGER_STREAM_SINK.write();
+        if guard.is_some() {
+            warn!(
+                "BindingLogger::set_stream_sink but already exist a sink, thus overriding. \
+                (This may or may not be a problem. It will happen normally if hot-reload Flutter app.)"
+            );
+        }
+        *guard = Some(stream_sink);
+        drop(guard)
+    }
+
+    fn record_to_entry(record: &Record) -> LogEntry {
+        LogEntry {
+            line: format!("{}", record.args()),
+            level: format!("{}", record.level()),
+        }
+    }
+}
+
+impl Log for DartLogger {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        metadata.level() <= max_level()
+    }
+
+    fn log(&self, record: &Record) {
+        if self.env_logger.enabled(record.metadata()) {
+            let entry = Self::record_to_entry(record);
+            if let Some(sink) = &*DART_LOGGER_STREAM_SINK.read() {
+                sink.add(entry);
+            }
+        }
+    }
+
+    fn flush(&self) {}
+}
+
+/* UniFFI */
+
+static INIT_UNIFFI_LOGGER: Once = Once::new();
+
+pub fn init_uniffi_logger(
+    log_stream: Box<dyn LogStream>,
+    filter_level: Option<BindingLevelFilter>,
+) {
+    INIT_UNIFFI_LOGGER.call_once(|| {
+        UniFFILogger::set_log_stream(log_stream, filter_level);
+    });
+}
+
+pub struct UniFFILogger {
```

Comment:
Why taking the specific loggers out of their specific crates?
Doesn't UniFFILogger belong to the sdk-bindings crate?

Reply:
Not necessarily, the loggers are not exposed through bindings.

I had brought them together to see the differences to find the root cause of the issue & I prefer having them together but we can bring them back to `uniffi_bindings.rs` & `bindings.rs` with little to no change.

Reply:
I agree putting these in the bindings crate seems logical

Reply:
Done.
- [Move Dart & UniFFI loggers back to their respective places](https://github.com/breez/breez-sdk-greenlight/pull/874/commits/8f661bb4cd812ad01068eefd95540b77dd2d9398)